test/unit/specs/utils/storageFactory.spec.js

```diff
@@ -10,5 +10,74 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-// eslint-disable-next-line no-unused-vars
 import storageFactory from "../../../../src/utils/storageFactory";
+
+describe("storageFactory", () => {
+  [
+    {
+      storageProperty: "session",
+      windowProperty: "sessionStorage"
+    },
+    {
+      storageProperty: "persistent",
+      windowProperty: "localStorage"
+    }
+  ].forEach(({ storageProperty, windowProperty }) => {
+    describe(storageProperty, () => {
+      describe("setItem", () => {
+        it("sets item", () => {
+          const window = {
+            [windowProperty]: {
+              setItem: jasmine.createSpy().and.returnValue(true)
+            }
+          };
+          const storage = storageFactory(window)("example.");
+          const result = storage[storageProperty].setItem("foo", "bar");
+          expect(window[windowProperty].setItem).toHaveBeenCalledWith(
+            "com.adobe.alloy.example.foo",
+            "bar"
+          );
+          expect(result).toBeTrue();
+        });
+
+        it("returns false if an error occurs setting item", () => {
+          const window = {
+            [windowProperty]: {
+              setItem: jasmine.createSpy().and.throwError()
+            }
+          };
+          const storage = storageFactory(window)("example.");
+          const result = storage[storageProperty].setItem("foo", "bar");
+          expect(result).toBeFalse();
+        });
+      });
+
+      describe("getItem", () => {
+        it("gets item", () => {
+          const window = {
+            [windowProperty]: {
+              getItem: jasmine.createSpy().and.returnValue("abc")
+            }
+          };
+          const storage = storageFactory(window)("example.");
+          const result = storage[storageProperty].getItem("foo");
+          expect(window[windowProperty].getItem).toHaveBeenCalledWith(
+            "com.adobe.alloy.example.foo"
+          );
+          expect(result).toBe("abc");
+        });
+
+        it("returns null if an error occurs while getting item", () => {
+          const window = {
+            [windowProperty]: {
+              getItem: jasmine.createSpy().and.throwError()
+            }
+          };
+          const storage = storageFactory(window)("example.");
+          const result = storage[storageProperty].getItem("foo");
+          expect(result).toBeNull();
+        });
+      });
+    });
+  });
+});
```

Comment:
This test is great, but somewhere we should have a test that makes sure the storageFactory works with the real browser APIs.